qa/rpc-tests/validateblocktemplate.py

```diff
@@ -0,0 +1,356 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2015 The Bitcoin Core developers
+# Copyright (c) 2015-2017 The Bitcoin Unlimited developers
+#
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+import pdb
+import sys
+if sys.version_info[0] < 3:
+    raise "Use Python 3"
+import logging
+logging.basicConfig(format='%(asctime)s.%(levelname)s: %(message)s', level=logging.INFO)
+
+# Test validateblocktemplate RPC call
+from test_framework.key import CECKey
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+from test_framework.script import *
+from test_framework.mininode import *
+from test_framework.blocktools import *
+
+# Create a transaction with an anyone-can-spend output, that spends the
+# nth output of prevtx.  pass a single integer value to make one output,
+# or a list to create multiple outputs
+
+
+def create_broken_transaction(prevtx, n, sig, value):
+    if not type(value) is list:
+        value = [value]
+    tx = CTransaction()
+    tx.vin.append(CTxIn(COutPoint(prevtx.sha256, n), sig, 0xffffffff))
+    for v in value:
+        tx.vout.append(CTxOut(v, b""))
+    tx.calc_sha256()
+    return tx
+
+
+def expectException(fn, ExcType, comparison=None):
+    try:
+        fn()
+    except ExcType as exc:
+        if comparison:
+            if comparison in str(exc):  # exception matchs
+                return
+            else:
+                print("Incorrect error.  Was: " + str(exc) + " Expecting: " + comparison)
+                assert(0)
+        else:
+            return
+    assert(0)  # an exception should have happened
+
+
+class ValidateblocktemplateTest(BitcoinTestFramework):
+
+    def setup_network(self):
+        self.nodes = []
+        self.nodes.append(start_node(0, self.options.tmpdir, ["-debug"]))
+        self.nodes.append(start_node(1, self.options.tmpdir, ["-debug"]))
+        self.is_network_split = False
+        connect_nodes(self.nodes[0], 1)
+
+    def run_test(self):
+        # Generate enough blocks to trigger certain block votes
+        self.nodes[0].generate(1001)
```

Comment:
no because the testing happens on node 0 until near the end of the test.  By not syncing until I need to, the test will find errors a lot quicker.